RyuSocks/Auth/UsernameAndPassword.cs

```diff
@@ -25,19 +28,48 @@ namespace RyuSocks.Auth
     [AuthMethodImpl(0x02)]
     public class UsernameAndPassword : IProxyAuth
     {
+        private readonly Dictionary<string, string> _database;
+
+        public UsernameAndPassword(Dictionary<string, string> database)
+        {
+            _database = database;
+        }
```

Comment:
Unfortunately you can't provide a constructor with parameters here.
The source generator expects auth method implementations to provide a parameterless constructor.

Could you instead expose a method for server-side initialization?

Reply:
fixed it cause i am cool